### PR TITLE
adapt filtering lambda to datasets 2.20

### PR DESCRIPTION
**Pull request by dafnapension**
Per issue #912

### CODE REVIEW

**Comment by dafnapension** [comment]
Hi @elronbandel ,  I do not manage to reproduce in my dev env. Even after cleaning my whole huggingface cache.   That is: all my local runs end successfully.
What do I miss?

**Comment by yoavkatz** [comment]
Maybe it was a local issue - I had?

**Comment by dafnapension** [comment]
Thanks, @yoavkatz. Yes, that was my puzzle: why is the regression failing, whereas my dev-env is laughing healthily all the way through?  And douze points!  my hf is indeed `datasets==2.19.0`.  I must comment, though, that `base.eqr` reads:  `datasets>=2.16.0`
Trying now with `2.20`, as your comment for the skip reads.

**Comment by dafnapension** [comment]
Hi @yoavkatz , did you notice this while upgrading to 2.20? perhaps this is one source of troubles?:

```
Installing collected packages: tqdm, requests, datasets
  Attempting uninstall: tqdm
    Found existing installation: tqdm 4.66.2
    Uninstalling tqdm-4.66.2:
      Successfully uninstalled tqdm-4.66.2
  Attempting uninstall: requests
    Found existing installation: requests 2.31.0
    Uninstalling requests-2.31.0:
      Successfully uninstalled requests-2.31.0
  Attempting uninstall: datasets
    Found existing installation: datasets 2.19.0
    Uninstalling datasets-2.19.0:
      Successfully uninstalled datasets-2.19.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
ibm-cos-sdk-core 2.13.4 requires urllib3<1.27,>=1.26.18; python_version < "3.10", but you have urllib3 2.2.1 which is incompatible.
Successfully installed datasets-2.20.0 requests-2.32.3 tqdm-4.66.4
```

File: src/unitxt/loaders.py [hunk 1]
```diff
@@ -177,18 +177,13 @@ def verify(self):
...
                 f"{self.__class__.__name__} cannot run use filtering_lambda expression without setting unitxt.settings.allow_unverified_code=True or by setting environment variable: UNITXT_ALLOW_UNVERIFIED_CODE."
             )
         logger.info(f"\nLoading filtered by: {self.filtering_lambda};")
-        return MultiStream(
-            {
-                name: dataset[name].filter(eval(self.filtering_lambda))
-                for name in dataset
-            }
-        )
+        return dataset.filter(eval(self.filtering_lambda))
```
Comment (elronbandel) [inline src/unitxt/loaders.py:186]: Should this line be protected by `allow_unverified_code` (https://github.com/IBM/unitxt/blob/6b1d93d2914e2551668c1803f59392411cee05c6/src/unitxt/settings_utils.py#L121)
@yoavkatz  can you advise?

**Review by elronbandel — CHANGES_REQUESTED** [review]

File: src/unitxt/loaders.py (hunk 1 above)
Comment (dafnapension) [reply src/unitxt/loaders.py:186]: Hi @elronbandel , it is well protected, in line 181, a few lines above the line you commented about

**Review by elronbandel — APPROVED** [review]

**Pull request merged**